node/src/tasks/bsp_upload_file_task.rs

```diff
@@ -100,22 +100,25 @@ where
         let file_key: HasherOutT<SHC::TrieLayout> = TryFrom::try_from(*event.file_key.as_ref())
             .map_err(|_| anyhow::anyhow!("File key and HasherOutT mismatch!"))?;
 
+        let proven = event
+            .chunk_with_proof
+            .proven
+            .first()
+            .ok_or_else(|| anyhow::anyhow!("Proven should contain one element; qed"))?;
```

Comment:
This could be multiple chunks too, maybe even required later for better upload speeds.

Reply:
Agreed, let's add a TODO here.